test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -137,7 +137,8 @@ public String getID() {
                                                                                       .put("segmentStoreReplicas", segmentStoreCount)
                                                                                       .put("debugLogging", true)
                                                                                       .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                                                                                      //.put("options", props.getProperties())
+                                                                                      .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
+                                                                                      .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
```

Comment:
Couple of questions: 
- Checked the pravega-operator and observe that https://github.com/pravega/pravega-operator/blob/0.3.2/pkg/apis/pravega/v1alpha1/pravega.go#L71
`DefaultSegmentStoreLimitMemory` is `2Gi`, this is the maximum memory that will be allotted to the pod/container. Now we are changing this to `5Gi`. 
The unintended side affect of this change would be that the pod will not be scheduled on the cluster if the request resource is not available on the cluster.

- The operator uses `-XUseCGroupMemoryLimitForHeap` and does not specify -Xmx. Will this cause issues with the rocksdb memory management which is outside the jvm process?

- Should we add memory constraints for the bookie or is the default enough for system tests?

Reply:
@shrids regarding the first point, I think that Kubernetes will schedule the pods based on the `requests`, not the `limits`, right? In this case, the request is `3Gi`, which is what should be available in a node so Kubernetes can spawn the pod.

Regarding the second point, the memory consumed by RocksDB is is classified as `OffHeap` memory of the Java process. To control the memory usage of RocksDB, we need to set right values to `rocksdb.writeBufferSizeMB` and `rocksdb.readCacheSizeMB` in `config-properties`, according to the memory provisioned to the SegmentStore process. I don't see any change in the default values of these properties for the Segment Store, so they are using very small default values (`writeBufferSizeMB=64`, `rocksdb.readCacheSizeMB=8`) that are not going to make the process to tun out of memory. 

From the last point, maybe we need to set more realistic configuration values to Pravega, so we test the system with a similar configuration compared to the one used by end-users.